multiversx/mx-chain-go
# Linter fixes for rc/v1.6.0 - 2023.08.30

## Reasoning behind the pull request
- linter issues
  
## Proposed changes
Implemented the following recommendations:
```
ioutil.ReadFile:  Deprecated: As of Go 1.16, this function simply calls os.ReadFile
ioutil.ReadAll:   Deprecated: As of Go 1.16, this function simply calls io.ReadAll
ioutil.TempDir:   Deprecated: As of Go 1.17, this function simply calls os.MkdirTemp
ioutil.WriteFile: Deprecated: As of Go 1.16, this function simply calls os.WriteFile
//+build comments can be removed
```

## Testing procedure
- standard system test, termui check to see the memory consumed by the process is not 0

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
